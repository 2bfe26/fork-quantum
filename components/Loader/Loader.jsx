import React from 'react';
import ContentLoader from 'react-content-loader';
import PropTypes from 'prop-types';
import LoaderTypes from './LoaderTypes';

/** Placeholder loader element
 */
class Loader extends React.Component {
  static Image = LoaderTypes.Image;
  static Text = LoaderTypes.Text;
  static List = LoaderTypes.List;
  static IconList = LoaderTypes.IconList;

  render() {
    const {
      children,
      primaryColor,
      secondaryColor,
      width,
      height,
      speed,
      ...rest
    } = this.props;

    return (
      <ContentLoader
        primaryColor={primaryColor}
        secondaryColor={secondaryColor}
        width={width}
        height={height}
        speed={speed}
        {...rest}
      >
        { children }
      </ContentLoader>);
  }
}

Loader.defaultProps = {
  primaryColor: '#ECEFF1',
  secondaryColor: '#CFD8DC',
  children: null,
  width: 400,
  height: 130,
  speed: 2,
};

Loader.propTypes = {
  primaryColor: PropTypes.string,
  secondaryColor: PropTypes.string,
  /** SVG to be rendered as a Loader */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  width: PropTypes.number,
  height: PropTypes.number,
  speed: PropTypes.number,
};

export default Loader;
