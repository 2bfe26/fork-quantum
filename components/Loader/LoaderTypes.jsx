import React from 'react';
import Loader from './Loader';

const types = {
  Image: props => (<Loader {...props} />),
  Text: props => (
    <Loader {...props}>
      <rect x="0" y="15" rx="3" ry="3" width="250" height="4" />
      <rect x="0" y="30" rx="3" ry="3" width="220" height="4" />
      <rect x="0" y="45" rx="3" ry="3" width="170" height="4" />
      <rect x="0" y="60" rx="3" ry="3" width="250" height="4" />
      <rect x="0" y="75" rx="3" ry="3" width="200" height="4" />
      <rect x="0" y="90" rx="3" ry="3" width="140" height="4" />
      <rect x="0" y="105" rx="3" ry="3" width="80" height="4" />
    </Loader>
  ),
  List: props => (
    <Loader {...props}>
      <circle cx="10" cy="15" r="4" />
      <rect x="25" y="13" rx="5" ry="5" width="220" height="4" />
      <circle cx="10" cy="40" r="4" />
      <rect x="25" y="38" rx="5" ry="5" width="220" height="4" />
      <circle cx="10" cy="65" r="4" />
      <rect x="25" y="63" rx="5" ry="5" width="220" height="4" />
      <circle cx="10" cy="90" r="4" />
      <rect x="25" y="88" rx="5" ry="5" width="220" height="4" />
      <circle cx="10" cy="115" r="4" />
      <rect x="25" y="113" rx="5" ry="5" width="220" height="4" />
    </Loader>
  ),
  IconList: props => (
    <Loader {...props}>
      <circle cx="15" cy="25" r="14" />
      <rect x="35" y="18" rx="4" ry="4" width="100" height="4" />
      <rect x="35" y="30" rx="4" ry="4" width="50" height="2" />

      <circle cx="15" cy="65" r="14" />
      <rect x="35" y="58" rx="4" ry="4" width="100" height="4" />
      <rect x="35" y="70" rx="4" ry="4" width="50" height="2" />

      <circle cx="15" cy="105" r="14" />
      <rect x="35" y="98" rx="4" ry="4" width="100" height="4" />
      <rect x="35" y="110" rx="4" ry="4" width="50" height="2" />
    </Loader>
  ),
};

function setDisplayName(name) {
  types[name].displayName = `Loader.${name}`;
}

Object
  .keys(types)
  .map(setDisplayName);

export default types;
