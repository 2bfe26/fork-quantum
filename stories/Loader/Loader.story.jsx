import React from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Loader from '../../components/Loader';
import LoaderTypes from '../../components/Loader/LoaderTypes';
import Heading from '../../.storybook/components/Heading';
import Atom from '../static/atom.svg';
import { TabbedView, Tab } from '../../.storybook/components/TabbedView';
import HowToImport from '../../.storybook/components/HowToImport';
import CodeExample from '../../.storybook/components/CodeExample';
import AutoPropsApi from '../../.storybook/components/AutoPropsApi';
import { Col, Row } from '../../components/Grid';

const LoaderExample = ({ name }) => {
  const Component = Loader[name] || Loader;
  const ExampleRow = styled(Row)`
    margin-bottom: 30px;
  `;

  return (
    <ExampleRow>
      <Col phone={4}>
        <CodeExample
          code={`<Loader${name ? `.${name}` : ''} />`}
          showTitle={false}
          component={Component}
        />
      </Col>
      <Col phone={4}>
        <Component />
      </Col>
    </ExampleRow>
  );
};

LoaderExample.defaultProps = { name: '' };
LoaderExample.propTypes = { name: PropTypes.string };

const loaderTypes = Object.keys(LoaderTypes);

storiesOf('12. Others', module)
  .add('Loader', () => (
    <Heading name="Loader" image={Atom}>
      <TabbedView>
        <Tab title="Usage">
          <HowToImport importModules="Loader" />
          <p>You can use these Loaders:</p>
          <ul>
            {loaderTypes.map(name => <li key={name}>{name}</li>)}
          </ul>

          <LoaderExample />

          {loaderTypes
            .map(type => <LoaderExample name={type} key={type} />)}

        </Tab>
        <Tab title="API">
          <AutoPropsApi component={Loader} />
        </Tab>
      </TabbedView>
    </Heading>
  ));
