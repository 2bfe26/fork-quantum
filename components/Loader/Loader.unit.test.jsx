import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './Loader';

describe('Loader component should match the snapshot', () => {
  it('Default', () => {
    expect(renderer.create(<Loader />).toJSON()).toMatchSnapshot();
  });
  it('Image', () => {
    expect(renderer.create(<Loader.Image />).toJSON()).toMatchSnapshot();
  });
  it('Text', () => {
    expect(renderer.create(<Loader.Text />).toJSON()).toMatchSnapshot();
  });
  it('List', () => {
    expect(renderer.create(<Loader.List />).toJSON()).toMatchSnapshot();
  });
  it('IconList', () => {
    expect(renderer.create(<Loader.IconList />).toJSON()).toMatchSnapshot();
  });
});
