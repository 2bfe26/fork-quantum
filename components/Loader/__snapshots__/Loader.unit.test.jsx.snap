// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match the snapshot Default 1`] = `
<svg
  className={undefined}
  preserveAspectRatio="xMidYMid meet"
  style={undefined}
  version="1.1"
  viewBox="0 0 400 130"
>
  <rect
    clipPath="url(#d610y4dv8hk)"
    height={130}
    style={
      Object {
        "fill": "url(#s56j8gryi4)",
      }
    }
    width={400}
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="d610y4dv8hk"
    >
      <rect
        height={130}
        rx="5"
        ry="5"
        width={400}
        x="0"
        y="0"
      />
    </clipPath>
    <linearGradient
      id="s56j8gryi4"
    >
      <stop
        offset="0%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#CFD8DC"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1.5; 1.5"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1; 2"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;

exports[`Loader component should match the snapshot IconList 1`] = `
<svg
  className={undefined}
  preserveAspectRatio="xMidYMid meet"
  style={undefined}
  version="1.1"
  viewBox="0 0 400 130"
>
  <rect
    clipPath="url(#izz5dj2jgal)"
    height={130}
    style={
      Object {
        "fill": "url(#oi7bek6yndc)",
      }
    }
    width={400}
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="izz5dj2jgal"
    >
      <circle
        cx="15"
        cy="25"
        r="14"
      />
      <rect
        height="4"
        rx="4"
        ry="4"
        width="100"
        x="35"
        y="18"
      />
      <rect
        height="2"
        rx="4"
        ry="4"
        width="50"
        x="35"
        y="30"
      />
      <circle
        cx="15"
        cy="65"
        r="14"
      />
      <rect
        height="4"
        rx="4"
        ry="4"
        width="100"
        x="35"
        y="58"
      />
      <rect
        height="2"
        rx="4"
        ry="4"
        width="50"
        x="35"
        y="70"
      />
      <circle
        cx="15"
        cy="105"
        r="14"
      />
      <rect
        height="4"
        rx="4"
        ry="4"
        width="100"
        x="35"
        y="98"
      />
      <rect
        height="2"
        rx="4"
        ry="4"
        width="50"
        x="35"
        y="110"
      />
    </clipPath>
    <linearGradient
      id="oi7bek6yndc"
    >
      <stop
        offset="0%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#CFD8DC"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1.5; 1.5"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1; 2"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;

exports[`Loader component should match the snapshot Image 1`] = `
<svg
  className={undefined}
  preserveAspectRatio="xMidYMid meet"
  style={undefined}
  version="1.1"
  viewBox="0 0 400 130"
>
  <rect
    clipPath="url(#9x2d98d0ph)"
    height={130}
    style={
      Object {
        "fill": "url(#ap62y21m2n7)",
      }
    }
    width={400}
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="9x2d98d0ph"
    >
      <rect
        height={130}
        rx="5"
        ry="5"
        width={400}
        x="0"
        y="0"
      />
    </clipPath>
    <linearGradient
      id="ap62y21m2n7"
    >
      <stop
        offset="0%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#CFD8DC"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1.5; 1.5"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1; 2"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;

exports[`Loader component should match the snapshot List 1`] = `
<svg
  className={undefined}
  preserveAspectRatio="xMidYMid meet"
  style={undefined}
  version="1.1"
  viewBox="0 0 400 130"
>
  <rect
    clipPath="url(#1s94d0cfrrqj)"
    height={130}
    style={
      Object {
        "fill": "url(#izfoo3fttf)",
      }
    }
    width={400}
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="1s94d0cfrrqj"
    >
      <circle
        cx="10"
        cy="15"
        r="4"
      />
      <rect
        height="4"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="13"
      />
      <circle
        cx="10"
        cy="40"
        r="4"
      />
      <rect
        height="4"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="38"
      />
      <circle
        cx="10"
        cy="65"
        r="4"
      />
      <rect
        height="4"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="63"
      />
      <circle
        cx="10"
        cy="90"
        r="4"
      />
      <rect
        height="4"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="88"
      />
      <circle
        cx="10"
        cy="115"
        r="4"
      />
      <rect
        height="4"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="113"
      />
    </clipPath>
    <linearGradient
      id="izfoo3fttf"
    >
      <stop
        offset="0%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#CFD8DC"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1.5; 1.5"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1; 2"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;

exports[`Loader component should match the snapshot Text 1`] = `
<svg
  className={undefined}
  preserveAspectRatio="xMidYMid meet"
  style={undefined}
  version="1.1"
  viewBox="0 0 400 130"
>
  <rect
    clipPath="url(#jf3ylks5tj)"
    height={130}
    style={
      Object {
        "fill": "url(#x7te71xk1l8)",
      }
    }
    width={400}
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="jf3ylks5tj"
    >
      <rect
        height="4"
        rx="3"
        ry="3"
        width="250"
        x="0"
        y="15"
      />
      <rect
        height="4"
        rx="3"
        ry="3"
        width="220"
        x="0"
        y="30"
      />
      <rect
        height="4"
        rx="3"
        ry="3"
        width="170"
        x="0"
        y="45"
      />
      <rect
        height="4"
        rx="3"
        ry="3"
        width="250"
        x="0"
        y="60"
      />
      <rect
        height="4"
        rx="3"
        ry="3"
        width="200"
        x="0"
        y="75"
      />
      <rect
        height="4"
        rx="3"
        ry="3"
        width="140"
        x="0"
        y="90"
      />
      <rect
        height="4"
        rx="3"
        ry="3"
        width="80"
        x="0"
        y="105"
      />
    </clipPath>
    <linearGradient
      id="x7te71xk1l8"
    >
      <stop
        offset="0%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#CFD8DC"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1.5; 1.5"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#ECEFF1"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="2s"
          repeatCount="indefinite"
          values="-1; 2"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;
